tools/evtool/evtindex_summary.cpp

```diff
@@ -0,0 +1,277 @@
+        out->put(headerSize, header);
+        StringBuffer line;
+        for (const Summary::value_type& s : summary)
+        {
+            line.append(s.first);
+            appendCSVColumn(line, fileInfo[s.first].str());
+            appendCSVColumn(line, s.second.branch.hits);
+            appendCSVColumn(line, s.second.branch.misses);
+            appendCSVColumn(line, s.second.branch.loads);
+            appendCSVColumn(line, s.second.branch.loadedSize);
+            appendCSVColumn(line, s.second.branch.evictions);
+            appendCSVColumn(line, s.second.branch.evictedSize);
+            appendCSVColumn(line, s.second.branch.elapsed);
+            appendCSVColumn(line, s.second.branch.read);
+            appendCSVColumn(line, s.second.leaf.hits);
+            appendCSVColumn(line, s.second.leaf.misses);
+            appendCSVColumn(line, s.second.leaf.loads);
+            appendCSVColumn(line, s.second.leaf.loadedSize);
+            appendCSVColumn(line, s.second.leaf.evictions);
+            appendCSVColumn(line, s.second.leaf.evictedSize);
+            appendCSVColumn(line, s.second.leaf.elapsed);
+            appendCSVColumn(line, s.second.leaf.read);
+            line.append('\n');
+            out->put(line.length(), line.str());
+            line.clear();
+        }
+    }
+
+public:
+    void setFile(const char *filename)
+    {
+        file.set(filename);
+    }
+
+    void setStream(IBufferedSerialOutputStream &out)
+    {
+        this->out.set(&out);
+    }
+
+    bool ready() const
+    {
+        return !file.isEmpty() && out.get();
+    }
+
+    bool summarize()
+    {
+        return readEvents(file, *this);
+    }
+
+protected:
+    struct NodeKindSummary;
+    NodeKindSummary& nodeSummary()
+    {
+        switch (currentNodeKind)
+        {
+        case 0: // branch
+            return summary[currentFileId].branch;
+        case 1: // leaf
+            return summary[currentFileId].leaf;
+        default:
+            throw makeStringExceptionV(-1, "unknown node kind: %llu", currentNodeKind);
+        }
+    }
+
+    void appendCSVColumn(StringBuffer& line, const char* value)
+    {
+        encodeCSVColumn(line.append(','), value);
+    }
+    void appendCSVColumn(StringBuffer& line, __uint64 value)
+    {
+        line.append(',').append(value);
+    }
+
+protected:
+    StringAttr file;
+    Linked<IBufferedSerialOutputStream> out;
+    using FileInfo = std::map<unsigned, StringAttr>;
+    struct NodeKindSummary
+    {
+        __uint64 hits{0};
+        __uint64 misses{0};
+        __uint64 loads{0};
+        __uint64 loadedSize{0};
+        __uint64 evictions{0};
+        __uint64 evictedSize{0};
+        __uint64 elapsed{0};
+        __uint64 read{0};
+    };
+    struct FileSummary
+    {
+        NodeKindSummary branch;
+        NodeKindSummary leaf;
+    };
+    using Summary = std::map<unsigned, FileSummary>;
+    FileInfo fileInfo;
+    Summary summary;
```

Comment:
minor: fileInfo and summary could be combined - if the filename was moved into the FileSummary struct.  Would be slightly cleaner.